src/io/flutter/run/test/DebugTestRunner.java

```diff
@@ -102,17 +103,21 @@ public void onTextAvailable(ProcessEvent event, Key outputType) {
             return;
           }
 
-          final String text = event.getText().trim();
+          final String text = event.getText();
           if (FlutterSettings.getInstance().isVerboseLogging()) {
-            LOG.info("[<-- " + text + "]");
+            LOG.info("[<-- " + text.trim() + "]");
           }
 
-          if (!text.startsWith("[{") || !text.endsWith("}]")) {
-            return; // Ignore anything not in our expected format.
-          }
+          stdoutParser.appendOutput(text);
 
-          final String json = text.substring(1, text.length() - 1);
-          dispatchJson(json);
+          for (String line : stdoutParser.getAvailableLines()) {
+            if (line.startsWith("[{")) {
+              line = line.trim();
```

Comment:
Is there any chance the json producer could break the stream mid-string? If so, this trim() could discard meaningful whitespace.

Reply:
I didn't document, but `getAvailableLines()` will return whole, concatenated lines (so, they will include the `[{...}]`.

Reply:
Thanks. I haven't read the parser code yet.